File: Part 2 - Regression/Simple Linear Regression/Simple Linear Regression codes/Simple_linear_regression.py
# -*- coding: utf-8 -*-
"""
Created on Tue May 23 12:18:54 2017

@author: pshrivas
"""

#==============================================================================
#Simple Linear Regression 
#==============================================================================
# Importing Libraries
# numpy includes mathematical tools
# matplotlib plot charts
# pandas import and manage datasets
 
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd 

#Importing Dataset

dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,1].values

#Splitting dataset into training set and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size = 1/3, random_state = 0)

#==============================================================================
# #Feature Scaling
# from sklearn.preprocessing import StandardScaler
# sc_X = StandardScaler()
# X_train = sc_X.fit_transform(X_train)
#  #for training we need to fit and then transform for test we have to only transform.
# X_test = sc_X.transform(X_test)
#==============================================================================
#==============================================================================
#Fitting Simple Linear Regression to the Training set 
#==============================================================================
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)


#==============================================================================
# Predicting the Test results
#==============================================================================
y_pred = regressor.predict(X_test)

#==============================================================================
# Visualising the Training set result
#==============================================================================
plt.scatter(X_train,y_train,color="red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.title('Salary vs Experience(Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('$Salary')
plt.show()

#==============================================================================
# Visualising the Test set result
#==============================================================================
plt.scatter(X_test,y_test,color="red")
plt.plot(X_train, regressor.predict(X_train), color = "blue")
plt.title('Salary vs Experience(Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('$Salary')
plt.show()

File: Part 2 - Regression/Simple Linear Regression/Simple Linear Regression codes/Simple_linear_regression.r
# Data PreProcessing

#Importing Dataset
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[,2:3]

#Splitting dataset into Training set and test set
# install.packages("caTools")
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary,SplitRatio = 2/3) #put split ratio for training set
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# #Feature Scaling
# 
# training_set[,2:3] = scale(training_set[,2:3]) #country and purchased is not numeric these are factors
# test_set[,2:3] = scale(test_set[,2:3])

#Simple Linear Regression

regressor = lm(formula = Salary ~ YearsExperience , data = training_set)

# summary(regressor)
# Coefficients:
#   Estimate Std. Error t value Pr(>|t|)    
# (Intercept)        25592       2646   9.672 1.49e-08 *** 
#   YearsExperience     9365        421  22.245 1.52e-14 *** -> high statistical significance
# 0 *-> no statistical significance
# ***->high statistical significance

y_pred <- predict(regressor,newdata = test_set)

#Visualising the Training set result
library(ggplot2)

ggplot()+
  geom_point(aes(x=training_set$YearsExperience,y=training_set$Salary),
             color="red") +
  geom_line(aes(x=training_set$YearsExperience, y = predict(regressor,newdata = training_set)),color="blue")+
  ggtitle('Salary vs Experience(Training set)')+xlab("Years of Experience")+ylab("salary")
  

ggplot()+
  geom_point(aes(x=test_set$YearsExperience,y=test_set$Salary), color="red")+
  geom_line(aes(training_set$YearsExperience,y=predict(regressor,newdata = training_set)),color="blue")+
  ggtitle('Salary vs Experience(Test set)')+xlab("Years of Experience")+ylab("salary")


